src/pages/finalize.js

```diff
@@ -143,17 +151,21 @@ export const finalizeTemplate = (data, specimenData) => {
     document.getElementById('contentBody').innerHTML = template;
     generateBarCode('connectIdBarCode', data.Connect_ID);
 
-    document.getElementById('finalizedSaveExit').addEventListener('click', () => {
+    document.getElementById('finalizedSaveExit') && document.getElementById('finalizedSaveExit').addEventListener('click', () => {
         searchTemplate();
     });
 
-    document.getElementById('finalizeForm').addEventListener('submit', (e) => { 
+    document.getElementById('returnToSpecimenSearch') && document.getElementById('returnToSpecimenSearch').addEventListener('click', () => {
+        collecionIdSearchScreenTemplate(appState.getState().username);
+    });
+
+    document.getElementById('finalizeForm') && document.getElementById('finalizeForm').addEventListener('submit', (e) => { 
         e.preventDefault();
         const modalId = e.target.getAttribute('data-modal-id');
         openModal(modalId);
     });
 
-    document.getElementById('finalizedConfirmButton').addEventListener('click', async (e) => { 
+    document.getElementById('finalizedConfirmButton') && document.getElementById('finalizedConfirmButton').addEventListener('click', async (e) => { 
         specimenData['410912345'] = 353358909;
         specimenData['556788178'] = new Date().toISOString();
```

Comment:
This is more for code readability for the future but these concept Ids can be referenced as well. [Field Mapping Reference](https://github.com/episphere/biospecimen/blob/dev/src/fieldToConceptIdMapping.js#L71C1-L94)